src/components/shared/SiweProviderConfig.tsx

```diff
@@ -0,0 +1,75 @@
+import { SIWEConfig } from "connectkit";
+import { SiweMessage } from "siwe";
+import { decodeJwt } from "jose";
+
+// TODO: this should probably be an environment variable
+const API_AUTH_PREFIX = "/api/v1/auth";
+const LOCAL_STORAGE_JWT_KEY = "agora-siwe-jwt";
+const AGORA_SIGN_IN = "Sign in to Agora with Ethereum";
+
+/* There's currently nothing stored on the backend to maintain session state.
+// All session state is stateless and stored in the JWT issued by the server.
+// Address, nonce, and chainId are all stored in the JWT, along with a particular
+// time to live/expiry.
+//
+// For signOut, the client should remove JWT from storage as applicable, and is otherwise
+// a no-op (pending AGORA-2015, or potential JWT-token tracking on our backend DB).
+//
+// JWT tokens for SIWE should therefore be issued with a short expiry time.
+*/
+
+const isSiweEnabled = () => {
+  return process.env.NEXT_PUBLIC_SIWE_ENABLED === "true";
+};
+
+export const siweProviderConfig: SIWEConfig = {
+  getNonce: async () =>
+    fetch(`${API_AUTH_PREFIX}/nonce`).then((res) => res.text()),
+  createMessage: ({ nonce, address, chainId }) =>
+    new SiweMessage({
+      version: "1",
+      domain: window.location.host,
+      uri: window.location.origin,
+      statement: AGORA_SIGN_IN,
+      address,
+      chainId,
+      nonce,
+    }).prepareMessage(),
+  verifyMessage: async ({ message, signature }) =>
+    fetch(`${API_AUTH_PREFIX}/verify`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        message,
+        signature,
+      }),
+    }).then(async (res) => {
+      // save JWT from verify to local storage
+      const token = await res.json();
+      localStorage.setItem(LOCAL_STORAGE_JWT_KEY, JSON.stringify(token));
+      return res.ok;
+    }),
+  getSession: async () => {
+    // return JWT from local storage
+    const session = localStorage.getItem(LOCAL_STORAGE_JWT_KEY);
```

Comment:
@stepandel fyi on this piece. the issued JWT maintains the user's session information, so currently this is stateless on the backend/server-side. Retrieving the session simply retrieves the JWT from local storage, and signing out removes it.

Reply:
let me know if you have strong opinions about this. the [connectkit documentation for SIWE](https://docs.family.co/connectkit/auth-with-custom-backend#siwe-custom-implementation-section-2-configure) is a little strange/misleading. The sample implementation of getSession seems to expect a `/session` endpoint:

```
const siweConfig: SIWEConfig = {
  ...
  getSession: async () => fetch('/api/siwe/session').then((res) => res.ok ? res.json() : null),
  ...
};
```

but does not provide any guidance about how the backend would resolve the session, given that nothing is provided as a parameter to that callback or route (neither in query nor path). Given that, I don't see how one would be able to resolve the session for the current wallet/user _without_ using client storage.

the documentation is also incorrect in that `onSignIn` and `onSignOut` are not members on SIWEConfig FWIW (see [here](https://github.com/family/connectkit/blob/release/v1.7.3/packages/connectkit/src/siwe/SIWEContext.tsx)). given that the documentation is a little wonky, I'm betting that they just didn't consider stateless session mgmt when writing this, so I think the approach in this PR should be okay for now.

Reply:
re: session endpoint. Even though JWT token contains all the data, it can only be decrypted on the server side. So we need this endpoint to decrypt and serve the data back to client.

Since it's gonna be used externally, only our server will be able to decrypt the token

Agree with you on the `onSignOut` `onSignIn` hooks. They're not necessary

Reply:
Encryption on the JWT is optional (see JWS vs JWE). So whether or not we encrypt is orthogonal. 

I'm amenable to encryption, but if we don't store anything private in the JWT then I'm also ok to keeping it unencrypted, if it saves us implementing /session

Either way, given the signature of the callback, we'll need to keep something in local storage to submit to the /session endpoint.

Reply:
oh interesting learning something new here! I'm okay with keeping it unencrypted in this case  as well then. This simplifies things -- nice!